vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -102,9 +102,10 @@ def pre_build(self):
         # want to raise an error if the column is not found in a figure's data_frame, it will just be ignored.
         # This is the case when bool(self.targets) is False.
         # Possibly in future this will change (which would be breaking change).
-        proposed_targets = self.targets or model_manager._get_page_model_ids_with_figure(
-            page_id=model_manager._get_model_page_id(model_id=ModelID(str(self.id)))
-        )
+        proposed_targets = self.targets or [
+            model.id
+            for model in model_manager._get_models((Graph, AgGrid, Table, Figure), model_manager._get_model_page(self))
```

Comment:
A little similar to before, but would we want to abstract this somewhere and have tuple as a constant, e.g. `FIGURE_MODELS`

Reply:
Good idea! Done, although it was a bit tricky with circular imports.